### PR TITLE
Move to GitHub Actions.

### PR DESCRIPTION
**Pull request by dvkruchinin**
### Motivation and context
Periodically reaching the job completion limit in Travis-ci.

### Checklist

- [x] I submit my changes into the `develop` branch
- [ ] I have added description of my changes into [CHANGELOG](https://github.com/opencv/cvat/blob/develop/CHANGELOG.md) file
- [ ] I have updated the [documentation](
  https://github.com/opencv/cvat/blob/develop/README.md#documentation) accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues ([read github docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary ([cvat-canvas](https://github.com/opencv/cvat/tree/develop/cvat-canvas#versioning),
[cvat-core](https://github.com/opencv/cvat/tree/develop/cvat-core#versioning), [cvat-data](https://github.com/opencv/cvat/tree/develop/cvat-data#versioning) and [cvat-ui](https://github.com/opencv/cvat/tree/develop/cvat-ui#versioning))

### License

- [x] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by azhavoro** [comment]
@nmanovic Could you please generate Coveralls repo token and set up `COVERALLS_REPO_TOKEN` secret for the CVAT repository, I have no permissions.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      - name: End-to-end testing
+        run: |
+          cd ./tests
+          npm install
+          npx cypress run --headless --browser chrome
+      - name: Collect coverage data
+        env:
+          HOST_COVERAGE_DATA_DIR: ${{ github.workspace }}
+          CONTAINER_COVERAGE_DATA_DIR: "/coverage_data"
+          COVERALLS_REPO_TOKEN: ${{ secrets.COVERALLS_REPO_TOKEN }}
```
Comment (azhavoro) [inline .github/workflows/main.yml:45]: @dvkruchinin There seems to be an issue with github secrets for PRs from forked repos:: https://docs.github.com/en/free-pro-team@latest/actions/reference/encrypted-secrets#using-encrypted-secrets-in-a-workflow

It should be possible to allow access to secrets from forked PRs, but this is not secure.
image

Another way is use `coverall action` to push coverage results https://github.com/marketplace/actions/coveralls-github-action. I'll to to configure.

Comment (dvkruchinin) [reply .github/workflows/main.yml:45]: I didn't expect this.
>  I'll try to configure.
Ok.

**Comment by azhavoro** [comment]
LGTM,
@bsekachev could you please review as well?

**Review by nmanovic — APPROVED** [review]

**Comment by nmanovic** [comment]
Need to add badge a badge: CI

**Pull request merged**